sw/device/lib/drivers/dma/dma_regs.h

```diff
@@ -51,6 +55,14 @@ extern "C" {
 #define DMA_DATA_TYPE_DATA_TYPE_OFFSET 0
 #define DMA_DATA_TYPE_DATA_TYPE_FIELD \
   ((bitfield_field32_t) { .mask = DMA_DATA_TYPE_DATA_TYPE_MASK, .index = DMA_DATA_TYPE_DATA_TYPE_OFFSET })
+#define DMA_DATA_TYPE_DATA_TYPE_VALUE_DMA_32BIT_WORD 0x0
+#define DMA_DATA_TYPE_DATA_TYPE_VALUE_DMA_16BIT_WORD 0x1
+#define DMA_DATA_TYPE_DATA_TYPE_VALUE_DMA_8BIT_WORD 0x2
+#define DMA_DATA_TYPE_DATA_TYPE_VALUE_DMA_8BIT_WORD_2 0x3
+
+// Restarts copying as soon as end of buffer is reach.
+#define DMA_CIRCULAR_MODE_REG_OFFSET 0x20
+#define DMA_CIRCULAR_MODE_CIRCULAR_MODE_BIT 0
```

Comment:
@JoseCalero This comment is unrelated to this PR, but this section of code is an excellent example. 

How the *_regs.h is generated makes major distinctions between different registers. 
So far I have seen these defines being added: 
* Offset: Bytes from the base address where this register is.
* Mask: Which bytes of the register belong to this field.
* Field offset: Bit in which the mask should be placed.
* Field: A structure containing the mask and field offset, so a value can be loaded just by passing the offset and this. 
* Bit: When the field is just one bit, which one. 
* Values: A set of possible values that this register can take.   

All this information is great BUT not every register has all of these defines. If a register is a 32 bit word, it wont need a field offset or mask, so it is not added. The Juan who wants generality hates this, especially because defines are free.

Anyways, the VALUE define is the one I find more interesting. I guess its generated from some comment on the RTL. This should be a MUST on every register with a limited number of possible values. Otherwise, we are defining them in the `module.h`. In the case of the dma trigger slots it is clear this should not be done there.

Reply:
we can talk about this and how to address it properly.
Anyways, as this is generated with the regtool (vendor from opentitan), we are not planning on changing that.